optics-core/src/Optics/Setter.hs

```diff
@@ -102,14 +102,28 @@ over o = \f -> runFunArrow $ getOptic (castOptic @A_Setter o) (FunArrow f)
 -- 'over' is used, because the first coordinate of a pair is never forced.
 --
 over'
-  :: Is k A_Setter
+  :: Is k A_Traversal
   => Optic k is s t a b
   -> (a -> b) -> s -> t
+-- See [Note: Solo wrapping]
 over' o = \f ->
-  let star = getOptic (castOptic @A_Setter o) $ Star (wrapIdentity' . f)
-  in unwrapIdentity' . runStar star
+  let star = getOptic (castOptic @A_Traversal o) $ Star ((Solo $!) . f)
+  in getSolo . runStar star
```

Comment:
Should we move `set'` and `over'` to `Optics.Traversal` instead?

Reply:
They would be a better fit, probably; I was just trying to avoid breaking things unnecessarily.